regression/cbmc/malloc-may-fail/test.desc

```diff
@@ -0,0 +1,9 @@
+CORE
+main.c
+--malloc-may-fail
+^EXIT=10$
+^SIGNAL=0$
+^\[main.assertion.\d+\] line \d+ assertion p: FAILURE$
+^VERIFICATION FAILED$
+--
+^warning: ignoring
```

Comment:
:pick: in future, we've found it helpful in test-gen to include a description explaining the purpose of the test below the line. for all tests. It helps when a test fails to understand why does it even exist. E.g

```
--
^warning: ignoring
--
Test that if --malloc-may-fail is enabled, that assert can return a null pointer
```